tests/functional/test_contract_instance.py

```diff
@@ -29,6 +29,17 @@ def test_init_specify_contract_type(
     assert contract.myNumber() == 2
 
 
+def test_call_using_block_identifier(
+    vyper_contract_instance, owner, chain, networks_connected_to_tester
+):
+    contract = vyper_contract_instance
+    contract.setNumber(1, sender=owner)
+    height = chain.blocks.height
+    contract.setNumber(33, sender=owner)
+    actual = contract.myNumber(block_identifier=height)
```

Comment:
This is very nice! One of the query types I eventually wanted to add was a historical query of view methods over a number of blocks